### PR TITLE
feat: (observability): trace Database.runTransactionAsync

### PR DESCRIPTION
**Pull request by odeke-em**
Extracted out of PR #2158, this change traces
Database.runTransactionAsync.

Updates #207

### CODE REVIEW

File: observability-test/spanner.ts [hunk 1]
```diff
@@ -468,6 +468,52 @@ describe('EndToEnd', () => {
       });
     });
 
+    it('runTransactionAsync', async () => {
+      const withAllSpansHaveDBName = generateWithAllSpansHaveDBName(
+        database.formattedName_
+      );
+      await database.runTransactionAsync(async transaction => {
+        const [rows] = await transaction!.run('SELECT 1');
```
Comment (alkatrivedi) [inline observability-test/spanner.ts:476]:
```suggestion
        await transaction!.run('SELECT 1');
```

File: src/database.ts [hunk 2]
```diff
@@ -3363,46 +3363,56 @@ class Database extends common.GrpcServiceObject {
...
+              session,
+              transaction,
+              runFn,
+              options
+            );
+
+            try {
+              return await runner.run();
+            } catch (e) {
+              setSpanError(span, e as Error);
```
Comment (surbhigarg92) [inline src/database.ts:3398]: use `setSpanErrorAndException`

**Review by surbhigarg92 — APPROVED** [review]

**Review by alkatrivedi — APPROVED** [review]

File: observability-test/spanner.ts (hunk 1 above)
Comment (odeke-em) [reply observability-test/spanner.ts]: Done, thanks!

File: src/database.ts (hunk 2 above)
Comment (odeke-em) [reply src/database.ts]: Nice catch, thank you!

**Pull request merged**